### ISSUE TITLE
phpcbf removes array elements

### ISSUE THREAD

**Issue by thiemowmde** [squizlabs/PHP_CodeSniffer#999]
Before:
```php
'props' => ['sitelinks', 'claims'],
```
After running `phpcbf` with the [mediawiki/mediawiki-codesniffer 0.7.1 rule set](https://github.com/wikimedia/mediawiki-tools-codesniffer/blob/v0.7.1/MediaWiki/ruleset.xml):
```php
'props' => [ 'sitelinks', ],
```
This happened in https://gerrit.wikimedia.org/r/#/c/287826/1/includes/SearchHookHandler.php, line 267. I can not provide more information. Please try to add a test that triggers this bug.

**Comment by aik099** [comment]
Are you sure, that built-in sniff do this? It can also be sniff created for MediaWiki standard itself.

**Comment by thiemowmde** [comment]
You are right, this is most probably something that only happens in combination with one of the custom MediaWiki sniffs. I did reported it there: https://phabricator.wikimedia.org/T134857. Please do not close this ticket before people are sure where the bug is.

**Comment by gsherwood** [comment]
I used this code snippet as a test:
```php
<?php
$params = wfArrayToCgi( array(
    'action' => 'wbgetentities',
    'props' => ['sitelinks', 'claims'],
    'format' => 'json',
    'ids' => implode( '|', $entityIds )
) );
```

I ran PHPCS over it using the MediaWiki standard and got a diff report with max verbosity. The important bit of the output is this:
```
---START FILE CONTENT---
1|<?php
2|$params = wfArrayToCgi( array(
3|    'action' => 'wbgetentities',
4|    'props' => ['sitelinks', 'claims'],
5|    'format' => 'json',
6|    'ids' => implode( '|', $entityIds )
7|) );
8|
--- END FILE CONTENT ---
	E: [Line 3] Tabs must be used to indent lines; spaces are not allowed (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
	Generic_Sniffs_WhiteSpace_DisallowSpaceIndentSniff (line 127) replaced token 11 (T_WHITESPACE) "····'action'" => "\t'action'"
	E: [Line 4] Tabs must be used to indent lines; spaces are not allowed (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
	Generic_Sniffs_WhiteSpace_DisallowSpaceIndentSniff (line 127) replaced token 19 (T_WHITESPACE) "····'props'" => "\t'props'"
	E: [Line 5] Tabs must be used to indent lines; spaces are not allowed (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
	Generic_Sniffs_WhiteSpace_DisallowSpaceIndentSniff (line 127) replaced token 32 (T_WHITESPACE) "····'format'" => "\t'format'"
	E: [Line 6] Tabs must be used to indent lines; spaces are not allowed (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
	Generic_Sniffs_WhiteSpace_DisallowSpaceIndentSniff (line 127) replaced token 40 (T_WHITESPACE) "····'ids'" => "\t'ids'"
	E: [Line 2] Short array syntax must be used to define arrays (Generic.Arrays.DisallowLongArraySyntax.Found)
	=> Changeset started by Generic_Sniffs_Arrays_DisallowLongArraySyntaxSniff (line 63)
		Q: Generic_Sniffs_Arrays_DisallowLongArraySyntaxSniff (line 68) replaced token 8 (T_ARRAY) "array" => ""
		Q: Generic_Sniffs_Arrays_DisallowLongArraySyntaxSniff (line 69) replaced token 9 (T_OPEN_PARENTHESIS) "(" => "["
		Q: Generic_Sniffs_Arrays_DisallowLongArraySyntaxSniff (line 70) replaced token 55 (T_CLOSE_PARENTHESIS) ")" => "]"
		A: Generic_Sniffs_Arrays_DisallowLongArraySyntaxSniff (line 73) replaced token 8 (T_ARRAY) "array" => ""
		A: Generic_Sniffs_Arrays_DisallowLongArraySyntaxSniff (line 73) replaced token 9 (T_OPEN_PARENTHESIS) "(" => "["
		A: Generic_Sniffs_Arrays_DisallowLongArraySyntaxSniff (line 73) replaced token 55 (T_CLOSE_PARENTHESIS) ")" => "]"
	=> Changeset ended: 3 changes applied
	W: Single space expected after opening parenthesis (MediaWiki.WhiteSpace.SpaceyParenthesis.SingleSpaceAfterOpenParenthesis)
	MediaWiki_Sniffs_WhiteSpace_SpaceyParenthesisSniff (line 118) replaced token 24 (T_OPEN_SHORT_ARRAY) "[" => "[·"
	W: Single space expected before closing parenthesis (MediaWiki.WhiteSpace.SpaceyParenthesis.SingleSpaceBeforeCloseParenthesis)
	MediaWiki_Sniffs_WhiteSpace_SpaceyParenthesisSniff (line 158) replaced token 28 (T_CONSTANT_ENCAPSED_STRING) "'claims'" => "·"
	* fixed 9 violations, starting loop 2 *
---START FILE CONTENT---
1|<?php
2|$params = wfArrayToCgi( [
3|	'action' => 'wbgetentities',
4|	'props' => [ 'sitelinks',  ],
5|	'format' => 'json',
6|	'ids' => implode( '|', $entityIds )
7|] );
8|
--- END FILE CONTENT ---
```
We can see that the sniff that removed `claims` was `MediaWiki_Sniffs_WhiteSpace_SpaceyParenthesisSniff` as it was trying to fix the warning `Single space expected before closing parenthesis`. It replaced this content with a single space character.

You'll need to report this with the MediaWiki standard project as they'll need to fix their sniff.

**Comment by gsherwood** [comment]
> Please do not close this ticket before people are sure where the bug is.

You can comment again if I'm wrong. I'll get notified. But I was unable to replicate this issue by using the MediaWiki standard without their custom sniffs, or by using any of the included standards.

**Comment by thiemowmde** [comment]
Thanks a lot. I updated https://phabricator.wikimedia.org/T134857.
